driver-core/src/main/java/com/datastax/driver/core/JdkSSLOptions.java

```diff
@@ -52,7 +53,12 @@ protected JdkSSLOptions(SSLContext context, String[] cipherSuites) {
 
     @Override
     public SslHandler newSSLHandler(SocketChannel channel) {
-        SSLEngine engine = newSSLEngine(channel);
+        return newSSLHandler(channel, null);
```

Comment:
I'm even wondering if this wouldn't be a showstopper actually. I think it would be safer to create subclasses, i.e. `RemoteEndpointAwareJdkSSLOptions extends JdkSSLOptions`.